### ISSUE TITLE
Problematic BRAT output due to complex event nesting.

### ISSUE THREAD

**Issue by hickst** [clulab/reach#383]
Need example of nested controller problem.

**Comment by myedibleenso** [comment]
@hickst, have you seen an example of this nesting in the web app's error log?

**Comment by hickst** [comment]
@myedibleenso No, but this may be an example:
`X up-regulates the Y-induced phosphorylation of Z.`

When I run this in the current BioVisualizer, no BRAT output is returned but no errors appear in any of the web app logs, either.

That is compared to this sentence, which I believe is a nested controlled:
`Ras mediates EGF activation of ERK.`

**Comment by hickst** [comment]
Hmmm...I take it back. The previous failure must be due to fake entities (x, y, z) because when I try this sentence `AKT up-regulates the KRAS-induced phosphorylation of p38.` I get output (albeit a bit odd-looking).

**Comment by myedibleenso** [comment]
This may have been addressed by the flattening, although currently the flattening is only performed when we produce "fries" output...

**Comment by myedibleenso** [comment]
Tests or it didn't happen...

**Comment by hickst** [comment]
No, that's bad software engineering. Fine to mark it unsolved and make it low priority but we should not be closing this.

**Comment by hickst** [comment]
GHP says:

> This sentence seems to break the visualizer:

At the same time, the catalytic activity of KSR1 was confirmed by the fact that tumour necrosis factor alpha (TNFα) and ceramide can induce KSR1 autophosphorylation and increase its capacity to phosphorylate and activate Raf-1

**Comment by danebell** [comment]
I think it breaks it because of the α character. When I replace same with `alpha`, it works.

**Comment by hickst** [comment]
@myedibleenso Yea, that sentence breaks my mind but it does seem to be an example of the type we were looking for previously: it breaks the BRAT visualization javascript. I believe the problem is these BRAT statements which have a relation as an element of an event:
```
E1	Positive_regulation:T1 Controlled:R1 Controller:T2
E2	Positive_regulation:T1 Controlled:R1 Controller:T3
E3	Positive_regulation:T1 Controlled:R1 Controller:T4
E4	Positive_regulation:T1 Controlled:R1 Controller:T5
```
I think the sentence can be simplified to "At the same time, the catalytic activity of KSR1 was confirmed by the fact that tumour necrosis factor alpha and ceramide can induce KSR1 autophosphorylation." and still break BRAT. Again, scanning the Event Annotations I see:

```E2	Positive_regulation:T2 Controlled:R1 Controller:T3```

which does not properly graph.

**Comment by danebell** [comment]
Oh, I see what you mean, now. Only the controllers get marked.

**Comment by hickst** [comment]
@danebell thanks for pointing that out but even if you eliminate the whole `(TNFa)` part, it still fails to graph some of the links. My theory is this is because of an event trying to control a relation (in BRAT).

**Comment by hickst** [comment]
BTW - there are absolutely no errors from the Visualizer backend (i.e., including Reach) and the Visualizer can actually graph the syntax tree for that twisted (and deeply nested) sentence.
